units/en/unitbonus3/rl-for-trading.mdx

```diff
@@ -0,0 +1,162 @@
+
+
+```python
+from gym_anytrading.datasets import FOREX_EURUSD_1H_ASK, STOCKS_GOOGL
+
+custom_env = gym.make(
+    'forex-v0',
+    df=FOREX_EURUSD_1H_ASK,
+    window_size=10,
+    frame_bound=(10, 300),
+    unit_side='right'
+)
+
+# custom_env = gym.make(
+#     'stocks-v0',
+#     df=STOCKS_GOOGL,
+#     window_size=10,
+#     frame_bound=(10, 300)
+# )
+```
+
+- It is to be noted that the first element of `frame_bound` should be greater than or equal to `window_size`.
+
+Remaining steps to create, train and further evaluate a Stable Baselines3 model are very similar to that as in other environments. 
+To follow along, here is a sample [notebook tutorial](https://github.com/AminHP/gym-anytrading/blob/master/examples/SB3_a2c_ppo.ipynb) available in Gym-Anytrading official GitHub repository.
+
+## FinRL for Trading
+[FinRL](https://github.com/AI4Finance-Foundation/FinRL) (Financial Reinforcement Learning) is an open-source framework designed to facilitate the application of RL in quantitative finance, particularly for automated stock trading. FinRL consists of three main layers: market environments, agents, and applications. This structure allows for the interaction between an agent and a market environment, enabling the agent to make sequential decisions.
+
+<figure>
+<img src="https://github.com/user-attachments/assets/60e0173b-c88a-4981-8da8-fb81e9f0e0b4" alt="FinRL Framework"/>
+<figcaption>Source: [FinRL GitHub repository](https://github.com/AI4Finance-Foundation/FinRL)</figcaption>
+</figure>
+
+1. **Market Environments: **FinRL provides gym-style market environments that simulate various stock markets, such as NASDAQ-100, DJIA, S&P 500, HSI, SSE 50, and CSI 300. These environments are configured with historical stock market datasets.
+2. **Agents: **The framework includes state-of-the-art deep reinforcement learning (DRL) algorithms like DQN, DDPG, PPO, SAC, A2C, and TD3. These agents are trained using neural networks to make trading decisions.
+3. **Applications: **FinRL supports various trading tasks, including single stock trading, multiple stock trading, and portfolio allocation. It also incorporates important trading constraints such as transaction costs, market liquidity, and risk aversion.
+
+### Example Workflow
+
+1. **Data Preparation:** Download historical stock data using libraries like Yahoo Finance API.
+
+```python
+from finrl.meta.preprocessor.yahoodownloader import YahooDownloader
+df = YahooDownloader(start_date='2009-01-01', end_date='2020-07-01', ticker_list=['AAPL', 'AMZN']).fetch_data()
+
+# Split df into df_train and df_test 
+```
+
+2. **Environment Configuration: **Create a trading environment using FinRL's StockTradingEnv class.
+
+```python
+from finrl.config import INDICATORS
+from finrl.meta.env_stock_trading.env_stocktrading import StockTradingEnv
+
+# This is only for illustration.
+# Values such as state_space, stock_dimension should be derived in the notebook before using them here.
+env_kwargs = {
```

Comment:
Isn't is simpler to give some mock (or default) values for all variables?

Reply:
Have made this clearer, hope this works.